src/lib/services/project-service.ts

```diff
@@ -1043,3 +1012,42 @@ export default class ProjectService {
         };
     }
 }
+
+const validateFlagNaming = (naming?: IFeatureNaming) => {
+    if (naming) {
+        const { pattern, example } = naming;
+        if (
+            pattern != null &&
+            example &&
+            !example.match(new RegExp(`^${pattern}$`))
+        ) {
+            throw new BadDataError(
+                `You've provided a feature flag naming example ("${example}") that doesn't match your feature flag naming pattern ("${pattern}"). Please provide an example that matches your supplied pattern. Bear in mind that the pattern must match the whole example, as if it were surrounded by '^' and "$".`,
+            );
+        }
+
+        if (!pattern && example) {
+            throw new BadDataError(
+                "You've provided a feature flag naming example, but no feature flag naming pattern. You must specify a pattern to use an example.",
+            );
+        }
+    }
+};
+
+// mutates the input object
+export const validateAndProcessFeatureNamingPattern = (
+    featureNaming: IFeatureNaming,
+): IFeatureNaming => {
+    featureNaming.pattern = featureNaming.pattern.replace(/(^\^+|\$+$)/g, '');
+
+    validateFlagNaming(featureNaming);
+
+    if (featureNaming.pattern && !featureNaming.example) {
+        featureNaming.example = null;
+    }
+    if (featureNaming.pattern && !featureNaming.description) {
+        featureNaming.description = null;
+    }
+
+    return featureNaming;
+};
```

Comment:
I've opted for mutating the input object here instead of creating a copy. That seems to be what we usually do, but I'd very happy to use a copy instead.